### PR TITLE
Big PR with multiple validations - [DO NOT MERGE YET]

### PR DESCRIPTION
**Pull request by ypriverol**
- [x] Change the logic of validation, by default everything is validated, and three variables are now introduced to skip some of the validations: 
    - [x] `--skip_ms_validation` 
    - [x] `--skip_factor_validation`
    - [x] `--skip_experimental_design_validation`  

Factor values columns validation: 
- [x] If factor values are not provided raise an error. 
- [x] If the factor values are absent in the `characteristics` or `comments` raise an error. 
- [x] If the values in the factor value column and their corresponding counterpart are different raise an error.  

Experimental design validation: 
- [x] If the data file and assay name combination are not unique raise an error.  
- [x] If `source name` has the same comment[technical replicate], comment[biological replicate], comment[label], comment[fraction identifier]

### CODE REVIEW

**Comment by ypriverol** [comment]
Related to issue: https://github.com/bigbio/quantms/issues/384

File: sdrf_pipelines/sdrf/sdrf.py [hunk 1]
```diff
@@ -77,3 +79,121 @@ def validate(self, template: str):
...
+            "comment[technical replicate]",
+            "characteristics[biological replicate]",
+            "comment[label]",
+            "comment[fraction identifier]",
+        ]
+
+        for col in cols:
+            if col not in self.columns:
+                error_message = (
+                    f"In order to perform experimental design validation, column '{col}' must be present in the SDRF"
```
Comment (jpfeuffer) [inline sdrf_pipelines/sdrf/sdrf.py:186]: I think that's a bit too restrictive. Wouldn't one assume that if e.g. no technical or biological replicate is given, that all of them are different ones?
Can we not imply this? Sdrfs are already verbose enough 😅

Comment (ypriverol) [reply sdrf_pipelines/sdrf/sdrf.py:186]: For an SDRF to be valid the technical and biological replicates MUST be provided.

**Review by jpfeuffer — APPROVED** [review]
Looks definitely safer than before.

**Review by daichengxin — APPROVED** [review]

**Pull request merged**